### DISCUSSION TITLE
How do I compile my external C++ scanner with -wall -werror?

### DISCUSSION THREAD

**Discussion by ahelwer** [tree-sitter/tree-sitter#1184] [Q&A] [answered]
After successfully shooting myself with as many C++ footguns as I could get my hands on, I think this would probably be a good idea. I'm assuming I need to make some modifications to the binding.gyp file; something like adding:
```
"cflags_cc": [
  "-wall", "-werror"
]
```
but this doesn't seem to work. I'm on Windows so using the msvc compiler, but ideally the solution I'm looking for would work on any platform. It should trigger whenever running `tree-sitter test` or `tree-sitter parse`.

**Comment by theHamsta** [comment, marked as answer]
`Tree-Sitter test` uses it's own C++ flags hard-coded in the rust binary. `Npm install` should build to the node bindings and show your warnings (I guess)

**Comment by ahlinc** [comment]
As @theHamsta said it's currently hard-coded. I suggested in https://github.com/tree-sitter/tree-sitter/issues/1246#issuecomment-876536579 how it can be solved by allowing to specify such flags in the `package.json` but it's the file related mostly to JS things and not all tree-sitter  users require JS or WASM bindings.

@maxbrunsfeld WDYT if to allow to specify compilation specific stuff right in the `grammar.js` file and the existence of external scanner also?
